src/data/index.ts

```diff
@@ -7,11 +7,32 @@ import { Memento } from 'vscode';
 import { PersistPartial } from 'redux-persist/es/persistReducer';
 import { persistedStateCodecNew } from '../persistedState/codecs';
 
-const buildStore = (workspaceState: Memento) => {
+const PERSISTANCE_PREFIX = 'persist';
+const PERSISTANCE_KEY = 'compressedRoot';
+
+const deserializeState = (serializedState: string) => {
+	const parsedState: Record<string, unknown> = {};
+
+	try {
+		const rawState = JSON.parse(serializedState);
+
+		Object.keys(rawState).forEach((key) => {
+			parsedState[key] = JSON.parse(rawState[key]);
+		});
+	} catch (e) {
+		console.log(e);
+	}
+
+	return parsedState;
+};
+
+const buildStore = async (workspaceState: Memento) => {
+	const storage = new MementoStorage(workspaceState);
+
 	const persistedReducer = persistReducer(
```

Comment:
do we need the `persistReducer` anymore? only for writes, am I right?

Reply:
we still need it, and we still need hydration action to be fired, otherwise persisting does not work. i think its because it relies on `persisted` flag, but i need to check it. in future we can think about removing this library at all and just subscribe to store and write directly to memento when soemthing chnages.